src/rt-app_parse_config.c

```diff
@@ -881,8 +881,17 @@ static sched_data_t *parse_sched_data(struct json_object *obj, int def_policy)
 		/* Support legacy grammar for thread object */
 		if (!tmp_data.runtime)
 			tmp_data.runtime = get_int_value_from(obj, "runtime", TRUE, 0);
-		if (!tmp_data.period)
-			tmp_data.period = get_int_value_from(obj, "period", TRUE, tmp_data.runtime);
+
+		if (!tmp_data.period) {
```

Comment:
This is only used to support legacy/old json file when the deadline parameters were define with the key "runtime", "deadline" and "period" and the "runtime" event didn't exist. The set  of deadline parameter is now done with dl-runtime, dl-deadline and dl-period and you must set them.
Your proposal tries to extend this backward compatibility to use the runtime and period key of some events but we will not do so.
a json file which wants to use deadline class must set the key dl-runtime, dl-deadline and dl-period. The only exception is for old json file that were using the old keys: runtime, deadline and period but not to try to estimate the value based on some events

Reply:
So in order to define a deadline task I need both (runtime, timer->period) and (dl-runtime, dl-period)?. For example:
```
{
  "tasks": {
    "task1": {
      "runtime": 30000,
      "timer": {
        "period": 65000,
        "mode": "absolute",
        "ref": "unique"
      },
      "dl-runtime": 30000,
      "dl-period": 65000
    },
    [...]
}
```
With maybe runtime = 0.95 * dl-runtime?

Reply:
Yes. you must set at least dl-runtime and dl-period. Then dl_deadline = dl_period if not set

Reply:
Perfect, then in order to avoid that the task misses its deadline I should use runtime < dl-runtime? let's say: runtime = 95% of dl-runtime.

One more request: could you please add at least one example that uses SCHED_DEADLINE with these parameters? It will be very helpful for who come next :)

Reply:
The ratio runtime vs dl-runtime is a bit platform and use case specific because it will depends if the runtime is in one run or split in several small ones as an example but 95% is probably a sane value. @jlelli  do you have any advice for this ratio ?
Please provide an example with deadline parameters that makes sense to you and I will merge it

Reply:
I prepared an example with utilization equal to 1.5 considering runtime = 95% of dl-runtime.
```
{
  "tasks": {
    "task2": {
      "runtime": 12059,
      "timer": {
        "period": 25000,
        "mode": "absolute",
        "ref": "unique"
      },
      "dl-runtime": 12694,
      "dl-period": 25000
    },
    "task1": {
      "runtime": 11108,
      "timer": {
        "period": 15000,
        "mode": "absolute",
        "ref": "unique"
      },
      "dl-runtime": 11693,
      "dl-period": 15000
    },
    "task3": {
      "runtime": 1863,
      "timer": {
        "period": 50000,
        "mode": "absolute",
        "ref": "unique"
      },
      "dl-runtime": 1962,
      "dl-period": 50000
    },
    "task4": {
      "runtime": 6970,
      "timer": {
        "period": 55000,
        "mode": "absolute",
        "ref": "unique"
      },
      "dl-runtime": 7337,
      "dl-period": 55000
    },
    "task0": {
      "runtime": 380,
      "timer": {
        "period": 10000,
        "mode": "absolute",
        "ref": "unique"
      },
      "dl-runtime": 400,
      "dl-period": 10000
    }
  },
  "global": {
    "duration": 4,
    "default_policy": "SCHED_DEADLINE",
    "pi_enabled": false,
    "lock_pages": false,
    "logdir": "./rtapp-log",
    "log_basename": "rt-app",
    "log_mode": "file",
    "gnuplot": true
  }
}
```

Reply:
> The ratio runtime vs dl-runtime is a bit platform and use case specific because it will depends if the runtime is in one run or split in several small ones as an example but 95% is probably a sane value. @jlelli do you have any advice for this ratio ?

Not much to add actually. Starting from 95% of dl-runtime seems a pretty good rule of thumb.